Remotion/Web/Core/PlainTextString.cs

```diff
@@ -0,0 +1,220 @@
+    /// <returns>
+    ///   <see langword="true" /> if both <paramref name="left"/> and <paramref name="right"/> have the same string value. Otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool operator == (PlainTextString left, PlainTextString right)
+    {
+      return left.Equals (right);
+    }
+
+    /// <summary>
+    /// Implements the inequality operator for <see cref="PlainTextString"/>s. The operator is implemented the same way as the <see cref="Equals(PlainTextString)"/> method.
+    /// </summary>
+    /// <param name="left">The first value to be compared for inequality.</param>
+    /// <param name="right">The second value to be compared for inequality.</param>
+    /// <returns>
+    ///   <see langword="true" /> if both <paramref name="left"/> and <paramref name="right"/> do not have the same string value. Otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool operator != (PlainTextString left, PlainTextString right)
+    {
+      return !left.Equals (right);
+    }
+
+    /// <summary>
+    /// Implements the implicit conversion from <see cref="PlainTextString"/> to <see cref="WebString"/>.
+    /// </summary>
+    /// <param name="value">The <see cref="PlainTextString"/> to be converted.</param>
+    /// <returns>A <see cref="WebString"/> of type <see cref="WebStringType"/>.<see cref="WebStringType.PlainText"/>.</returns>
+    public static implicit operator WebString (PlainTextString value)
+    {
+      return WebString.CreateFromText (value._value);
+    }
+
+    /// <summary>
+    /// Implements the explicit conversion from <see cref="string"/> to <see cref="PlainTextString"/>.
+    /// </summary>
+    /// <param name="value">The <see cref="string"/> to be converted.</param>
+    /// <returns>A <see cref="PlainTextString"/> with the specified <paramref name="value"/>.</returns>
+    public static explicit operator PlainTextString (string value)
+    {
+      return new PlainTextString (value);
+    }
+
+    [CanBeNull]
+    private readonly string? _value;
+
+    private PlainTextString ([CanBeNull] string? value)
+    {
+      _value = value;
+    }
+
+    /// <summary>
+    /// Tests whether the <see cref="PlainTextString"/>s value is <see langword="null" /> or an empty string.
+    /// </summary>
+    public bool IsEmpty => string.IsNullOrEmpty (_value);
+
+    /// <summary>
+    /// Gets the raw string value of the <see cref="PlainTextString"/> or an empty string if there is no value.
+    /// </summary>
+    [NotNull]
+    public string GetValue () => _value ?? string.Empty;
+
+    /// <summary>
+    /// Writes the <see cref="PlainTextString"/>'s value to the specified <paramref name="writer"/>.
+    /// The value is escaped before being written.
+    /// </summary>
+    /// <param name="writer">The <see cref="HtmlTextWriter"/> where the value will be appended to. Must not be <see langword="null" />.</param>
+    public void Write ([NotNull] HtmlTextWriter writer)
```

Comment:
I think it would be a good idea to delegate to WebString for the various rendering methods. Otherwise we do have duplication and PlainTextString is kind of a special case of WebString. Once we follow up with the idea of making these into HtmlTextWriter-extensions, the problem of where to locate them gets solved anyway.